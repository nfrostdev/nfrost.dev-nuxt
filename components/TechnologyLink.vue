<template>
  <a :href="technology.url" target="_blank" rel="noopener noreferrer" class="technology-link">{{ technology.title }}</a>
</template>

<script lang="ts">
  import Vue from 'vue'

  export default Vue.extend({
    props: {
      technology: Object
    },
    mounted() {
      console.log(this.technology);
    }
  })
</script>

<style lang="scss">
  .technology-link {
    @apply inline-block bg-indigo-200 leading-none p-2 rounded-sm transition duration-200 ease-in-out text-sm;

    @media(prefers-color-scheme: dark) {
      @apply bg-blue-900;
    }

    &:hover, &:focus {
      @apply bg-indigo-300;

      @media(prefers-color-scheme: dark) {
        @apply bg-blue-800;
      }
    }
  }
</style>
