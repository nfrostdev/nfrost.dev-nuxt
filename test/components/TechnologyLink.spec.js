import {shallowMount} from '@vue/test-utils'
import TechnologyLink from '../../components/TechnologyLink'

const factory = () => {
  return shallowMount(TechnologyLink, {
    propsData: {
      technology: {
        url: 'https://www.google.com/',
        title: 'Google'
      }
    }
  });
};

describe('TechnologyLink', () => {
  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  })
  test('has link', () => {
    const wrapper = factory();
    expect(wrapper.html()).toContain('<a href="https://www.google.com/" target="_blank" rel="noopener noreferrer" class="technology-link">Google</a>');
  })
})
