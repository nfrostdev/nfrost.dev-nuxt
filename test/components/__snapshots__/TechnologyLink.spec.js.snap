// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechnologyLink renders properly 1`] = `"<a href=\\"https://www.google.com/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"technology-link\\">Google</a>"`;
